core/controllers/domain_objects_validator.py

```diff
@@ -96,3 +98,21 @@ def validate_collection_change(obj):
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
     collection_domain.CollectionChange(obj)
+
+
+def validate_email_dashboard_data(data):
```

Comment:
@rohitkatlaa I am concerned that -- this validation is written in the handler class (not a domain validator), I have moved that validator to here. 
Looking forward for your suggestions.

Reply:
I think this is better. Thanks.